src/services/http-req/dogBreedRequest.ts

```diff
@@ -0,0 +1,32 @@
+import fetch from 'node-fetch'
+import AbortController from 'abort-controller'
+import { Status, RequestedDogBreeds } from '../../lambdas/types'
+
+export async function getDogBreeds(): Promise<RequestedDogBreeds> {
+  const controller = new AbortController()
+  let reqTimedOut = false
+  const timeout = setTimeout(() => {
+    controller.abort()
+    reqTimedOut = true
+  }, 1500)
+
+  try {
+    const res = await fetch('https://dog.ceo/api/breeds/list/all', { signal: controller.signal })
+    const payload = await res.json()
+    clearTimeout(timeout)
+
+    if (reqTimedOut) {
+      throw new Error('Request to externial API timed out')
+    }
```

Comment:
I saw that too but I could not get it to work in the code for some reason. I also tried the node-abort-controller package which is supposed to be a polyfill for node. I am not sure what I am missing, because I tried adding the example code from node-fetch and node-abort-controller and running it but I still couldn't get an error either in the function or in the test suite running it. I guess this is my work around right now, but if you see something I missed I would really like to be able to get this to work as expected.

Reply:
I left some comments around this, hopefully that points you in the right direction. If you're still not getting it to work you can reach out here and I can answer any questions you might have

Reply:
thanks, yes those comments were the aha moment of what I was missing